app/models/sushi.rb

```diff
@@ -156,4 +156,22 @@ def coerce_metric_types(params = {})
                       end
     end
   end
+
+  # This param specifies the granularity of the usage data to include in the report.
+  # Permissible values are Month (default) and Totals.
+  # For Totals, each Item_Performance element represents the aggregated usage for the reporting period.
+  # See https://cop5.projectcounter.org/en/5.1/03-specifications/03-counter-report-common-attributes-and-elements.html#report-filters-and-report-attributes for details
+  module GranularityCoercion
+    extend ActiveSupport::Concern
+    included do
+      attr_reader :granularity_string, :granularity_in_params
+    end
+    ALLOWED_GRANULARITY = ["Month", "Totals"].freeze
+
+    def coerce_granularity(params = {})
+      @granularity_in_params = params.key?(:granularity) &&
+                               ALLOWED_GRANULARITY.any? { |allowed_granularity| allowed_granularity.casecmp(params[:granularity].downcase).zero? }
```

Comment:
This might be a bit less hoops for 

```ruby
@granularity_in_params = ALLOWED_GRANULARITY.include?(params[:granularity].to_s.titleize)
```

Reply:
@jeremyf just made the updates we talked about ･ᴗ･